packages/wonder-blocks-modal/components/modal-dialog.js

```diff
@@ -13,17 +17,18 @@ export default class ModalDialog extends React.Component<Props> {
     render() {
         const {style, children} = this.props;
         return (
-            <MediaLayout
-                spec={MEDIA_MODAL_SPEC}
-                style={[
-                    styles.wrapper,
-                    (mediaSize) => mediaSize === "small" && styles.small,
-                    style,
-                ]}
-                role="dialog"
-                aria-labelledby="wb-modal-title"
-            >
-                {children}
+            <MediaLayout spec={MEDIA_MODAL_SPEC}>
+                <View
+                    style={[
+                        styles.wrapper,
+                        (mediaSize) => mediaSize === "small" && styles.small,
```

Comment:
I just realized that this means that `mediaSize` can only be use to change the `style` but devs may also want to use `mediaSize` to toggle whether a component is rendered at all or to perform other logic.  I think we should have a component that provides the `mediaSize` (and the `mediaSpec`) using the function-as-children pattern.  I'll put together a separate PR and see what you think.

Reply:
Also, in talking with other devs, people in general preferred the FaC pattern as opposed to the HoC pattern, mainly due to its flexibility.

Reply:
One additional though about this, allow callbacks to be passed `style` makes our `style` incompatible with React Native's `style` prop which was one of the reasons for adopting our current `style` prop in the first place.

Sorry for not thinking about these arguments when this callback was first introduced.

Reply:
Nope, I though of this too! Which is why I wrote up the MediaLayout/Layout spec - I'll be pursuing that next! https://docs.google.com/document/d/1hdh0_tEn0RY7jkC71Sln2BS3xJ9alwFzW6qYRiKu3kU/edit

Reply:
Yes!  This is much better.  Thanks for iterating on this. :)

Reply:
I totally forgot about that doc. 😅